### ISSUE TITLE
The clients in Azure SDK for JS use TokenCredential to construct instead of TokenCredentialBase

### ISSUE THREAD

**Issue by zhongyi-zhang** [microsoft/vscode-azure-account#317]
For example arm source client: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/resources/arm-resources/src/resourceManagementClient.ts#L41

The extension returns credential by `TokenCredentialsBase | AzureIdentityCredentialAdapter`:  https://github.com/microsoft/vscode-azure-account/blob/main/src/azure-account.api.d.ts#L39.

So if installing the latest Azure SDK for JS, we could hit the issue
```
Argument of type 'TokenCredentialsBase | AzureIdentityCredentialAdapter' is not assignable to parameter of type 'TokenCredential'.
  Type 'TokenCredentialsBase' is not assignable to type 'TokenCredential'.
    The types returned by 'getToken(...)' are incompatible between these types.
      Type 'Promise<TokenResponse>' is not assignable to type 'Promise<AccessToken | null>'.
        Type 'TokenResponse' is missing the following properties from type 'AccessToken': token, expiresOnTimestampts(2345)
```
in client construction.

Any solution? Or does the extension need an update? Thanks.

**Comment by zhongyi-zhang** [comment]
The `@azure/ms-rest-nodeauth` from https://github.com/Azure/azure-sdk-for-node is deprecated.

>We are excited to announce that we are building a new Azure SDK for JavaScript in a new GitHub repo located here: https://github.com/azure/azure-sdk-for-js. These packages ship with TypeScript type definitions and have both Node.js and Browser support.

The repo is not updated for months.

**Comment by wwlorey** [comment]
We should switch all instances of `TokenCredentialsBase` to `TokenCredential` (from @azure/core-auth).

**Comment by wwlorey** [comment]
[zhongyi-zhang](https://github.com/zhongyi-zhang) we shipped support for `TokenCredential` in [v0.10.0](https://github.com/microsoft/vscode-azure-account/releases/tag/v0.10.0). Thanks for filing this!

**Comment by zhongyi-zhang** [comment]
@wwlorey wow, glad to see that `TokenCredentialBase` is still returned for back forward compatibility.  So that the extensions which depend on its apis do not break.
